File: 204. Count Primes/main.py
class Solution:
    def countPrimes(self, n: int) -> int:
        is_primes = [True] * n
        count = 0

        for i in range(2, n):
            if not is_primes[i]:
                continue
            count += 1

            for mult in range(i * i, n, i):
                is_primes[mult] = False

        return count


class Solution:
    def countPrimes(self, n: int) -> int:
        is_primes = [True] * n
        count = 0

        for i in range(2, n):
            if is_primes[i]:
                count += 1
                is_primes[i * i : n : i] = [False] * len(is_primes[i * i : n : i])

        return count

File: 204. Count Primes/main.rs
struct Solution;

impl Solution {
    pub fn count_primes(n: i32) -> i32 {
        let mut is_primes = vec![true; n as usize];
        let mut count = 0;

        for i in 2..n as usize {
            if is_primes[i] {
                count += 1;
                for j in (i * i..n as usize).step_by(i) {
                    is_primes[j] = false;
                }
            }
        }

        count
    }
}
